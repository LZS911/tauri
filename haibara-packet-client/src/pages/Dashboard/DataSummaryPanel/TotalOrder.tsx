import { useTranslation } from 'react-i18next';
import { DashboardPanelEnum } from '..';
import SummaryPanelWrapper, { DataTrendEnum } from '../Common';

const mockData = () => {
  return {
    trend: DataTrendEnum.fall,
    trendData: 0.224,
    totalData: 17364,
  };
};

const TotalOrder: React.FC = () => {
  const { t } = useTranslation();

  return (
    <SummaryPanelWrapper
      key={DashboardPanelEnum.TotalOrder}
      title={t('backend.dashboard.dataSummaryPanel.title3')}
      {...mockData()}
    />
  );
};

export default TotalOrder;
