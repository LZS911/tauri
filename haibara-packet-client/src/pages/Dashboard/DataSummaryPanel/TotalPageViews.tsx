import { useTranslation } from 'react-i18next';
import { DashboardPanelEnum } from '..';
import SummaryPanelWrapper, { DataTrendEnum } from '../Common';

const mockData = () => {
  return {
    trend: DataTrendEnum.rise,
    trendData: 0.627,
    totalData: 43298,
  };
};

const TotalPageViews: React.FC = () => {
  const { t } = useTranslation();

  return (
    <SummaryPanelWrapper
      key={DashboardPanelEnum.TotalPageViews}
      title={t('backend.dashboard.dataSummaryPanel.title1')}
      {...mockData()}
    />
  );
};

export default TotalPageViews;
