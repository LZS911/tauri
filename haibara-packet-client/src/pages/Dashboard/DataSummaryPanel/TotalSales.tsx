import { useTranslation } from 'react-i18next';
import { DashboardPanelEnum } from '..';
import SummaryPanelWrapper, { DataTrendEnum } from '../Common';

const mockData = () => {
  return {
    trend: DataTrendEnum.fall,
    trendData: 0.627,
    totalData: 35078,
  };
};

const TotalSales: React.FC = () => {
  const { t } = useTranslation();

  return (
    <SummaryPanelWrapper
      key={DashboardPanelEnum.TotalSales}
      title={t('backend.dashboard.dataSummaryPanel.title4')}
      addSymbol={true}
      {...mockData()}
    />
  );
};

export default TotalSales;
