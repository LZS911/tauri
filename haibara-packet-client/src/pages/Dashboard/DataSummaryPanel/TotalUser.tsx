import { useTranslation } from 'react-i18next';
import { DashboardPanelEnum } from '..';
import SummaryPanelWrapper, { DataTrendEnum } from '../Common';

const mockData = () => {
  return {
    trend: DataTrendEnum.rise,
    trendData: 0.717,
    totalData: 76221,
  };
};

const TotalUsers: React.FC = () => {
  const { t } = useTranslation();

  return (
    <SummaryPanelWrapper
      key={DashboardPanelEnum.TotalUsers}
      title={t('backend.dashboard.dataSummaryPanel.title2')}
      {...mockData()}
    />
  );
};

export default TotalUsers;
