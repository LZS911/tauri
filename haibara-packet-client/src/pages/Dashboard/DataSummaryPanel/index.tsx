import { useTranslation } from 'react-i18next';
import TotalOrder from './TotalOrder';
import TotalPageViews from './TotalPageViews';
import TotalSales from './TotalSales';
import TotalUser from './TotalUser';

const DataSummaryPanel = () => {
  const { t } = useTranslation();
  return (
    <>
      <h2 className="font-semibold dark:text-white">
        {t('dashboard.dataSummary.title')}
      </h2>
      <div className="mb-6 flex flex-wrap justify-between">
        <TotalPageViews />
        <TotalUser />
        <TotalOrder />
        <TotalSales />
      </div>
    </>
  );
};

export default DataSummaryPanel;
