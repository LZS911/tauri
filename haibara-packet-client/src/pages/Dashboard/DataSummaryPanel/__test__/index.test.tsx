import { render } from '@testing-library/react';
import DataSummaryPanel from '../index';
import TotalOrder from '../TotalOrder';
import TotalPageViews from '../TotalPageViews';
import TotalSales from '../TotalSales';
import TotalUser from '../TotalUser';

describe('test DataSummaryPanel', () => {
  test('should match snapshot', () => {
    const { container } = render(<DataSummaryPanel />);
    expect(container).toMatchSnapshot();
  });
});

describe('test TotalOrder', () => {
  test('should match snapshot', () => {
    const { container } = render(<TotalOrder />);
    expect(container).toMatchSnapshot();
  });
});

describe('test TotalPageViews', () => {
  test('should match snapshot', () => {
    const { container } = render(<TotalPageViews />);
    expect(container).toMatchSnapshot();
  });
});

describe('test TotalSales', () => {
  test('should match snapshot', () => {
    const { container } = render(<TotalSales />);
    expect(container).toMatchSnapshot();
  });
});

describe('test TotalUser', () => {
  test('should match snapshot', () => {
    const { container } = render(<TotalUser />);
    expect(container).toMatchSnapshot();
  });
});
