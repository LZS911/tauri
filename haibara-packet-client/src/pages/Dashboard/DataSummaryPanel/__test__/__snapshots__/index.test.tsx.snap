// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test DataSummaryPanel should match snapshot 1`] = `
<div>
  <h2
    class="font-semibold dark:text-white"
  >
    dashboard.dataSummary.title
  </h2>
  <div
    class="mb-6 flex flex-wrap justify-between"
  >
    <div
      class="bg-white dark:bg-darkMode dark:text-white mt-4 max-w-[100%] basis-[100%] rounded-md border border-stone-200 p-5 dark:border-stone-700 md:max-w-[48%] md:basis-[48%] lg:max-w-[32%] lg:basis-[32%] xl:max-w-[23%] xl:basis-[23%]"
    >
      <div
        class="opacity-70 hover:cursor-pointer"
      >
        backend.dashboard.dataSummaryPanel.title1
      </div>
      <div
        class="mt-2 mb-4 flex"
      >
        <div
          class="mr-4 text-xl font-bold"
        >
          43,298
        </div>
        <span
          class="border-primary bg-secondary text-primary flex items-center rounded-md border  px-2 text-xs"
        >
          <span
            aria-label="rise"
            class="anticon anticon-rise mr-1"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="rise"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M917 211.1l-199.2 24c-6.6.8-9.4 8.9-4.7 13.6l59.3 59.3-226 226-101.8-101.7c-6.3-6.3-16.4-6.2-22.6 0L100.3 754.1a8.03 8.03 0 000 11.3l45 45.2c3.1 3.1 8.2 3.1 11.3 0L433.3 534 535 635.7c6.3 6.2 16.4 6.2 22.6 0L829 364.5l59.3 59.3a8.01 8.01 0 0013.6-4.7l24-199.2c.7-5.1-3.7-9.5-8.9-8.8z"
              />
            </svg>
          </span>
          62.7%
        </span>
      </div>
      <div
        class="overflow-hidden text-ellipsis whitespace-nowrap text-xs opacity-70"
      >
        dashboard.dataSummary.panelRiseDesc
      </div>
    </div>
    <div
      class="bg-white dark:bg-darkMode dark:text-white mt-4 max-w-[100%] basis-[100%] rounded-md border border-stone-200 p-5 dark:border-stone-700 md:max-w-[48%] md:basis-[48%] lg:max-w-[32%] lg:basis-[32%] xl:max-w-[23%] xl:basis-[23%]"
    >
      <div
        class="opacity-70 hover:cursor-pointer"
      >
        backend.dashboard.dataSummaryPanel.title2
      </div>
      <div
        class="mt-2 mb-4 flex"
      >
        <div
          class="mr-4 text-xl font-bold"
        >
          76,221
        </div>
        <span
          class="border-primary bg-secondary text-primary flex items-center rounded-md border  px-2 text-xs"
        >
          <span
            aria-label="rise"
            class="anticon anticon-rise mr-1"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="rise"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M917 211.1l-199.2 24c-6.6.8-9.4 8.9-4.7 13.6l59.3 59.3-226 226-101.8-101.7c-6.3-6.3-16.4-6.2-22.6 0L100.3 754.1a8.03 8.03 0 000 11.3l45 45.2c3.1 3.1 8.2 3.1 11.3 0L433.3 534 535 635.7c6.3 6.2 16.4 6.2 22.6 0L829 364.5l59.3 59.3a8.01 8.01 0 0013.6-4.7l24-199.2c.7-5.1-3.7-9.5-8.9-8.8z"
              />
            </svg>
          </span>
          71.7%
        </span>
      </div>
      <div
        class="overflow-hidden text-ellipsis whitespace-nowrap text-xs opacity-70"
      >
        dashboard.dataSummary.panelRiseDesc
      </div>
    </div>
    <div
      class="bg-white dark:bg-darkMode dark:text-white mt-4 max-w-[100%] basis-[100%] rounded-md border border-stone-200 p-5 dark:border-stone-700 md:max-w-[48%] md:basis-[48%] lg:max-w-[32%] lg:basis-[32%] xl:max-w-[23%] xl:basis-[23%]"
    >
      <div
        class="opacity-70 hover:cursor-pointer"
      >
        backend.dashboard.dataSummaryPanel.title3
      </div>
      <div
        class="mt-2 mb-4 flex"
      >
        <div
          class="mr-4 text-xl font-bold"
        >
          17,364
        </div>
        <span
          class="border-yellow-500 bg-[#fff7e0] text-yellow-500 dark:bg-[#342c1a] flex items-center rounded-md border  px-2 text-xs"
        >
          <span
            aria-label="fall"
            class="anticon anticon-fall mr-1"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="fall"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M925.9 804l-24-199.2c-.8-6.6-8.9-9.4-13.6-4.7L829 659.5 557.7 388.3c-6.3-6.2-16.4-6.2-22.6 0L433.3 490 156.6 213.3a8.03 8.03 0 00-11.3 0l-45 45.2a8.03 8.03 0 000 11.3L422 591.7c6.2 6.3 16.4 6.3 22.6 0L546.4 490l226.1 226-59.3 59.3a8.01 8.01 0 004.7 13.6l199.2 24c5.1.7 9.5-3.7 8.8-8.9z"
              />
            </svg>
          </span>
          22.4%
        </span>
      </div>
      <div
        class="overflow-hidden text-ellipsis whitespace-nowrap text-xs opacity-70"
      >
        dashboard.dataSummary.panelFailDesc
      </div>
    </div>
    <div
      class="bg-white dark:bg-darkMode dark:text-white mt-4 max-w-[100%] basis-[100%] rounded-md border border-stone-200 p-5 dark:border-stone-700 md:max-w-[48%] md:basis-[48%] lg:max-w-[32%] lg:basis-[32%] xl:max-w-[23%] xl:basis-[23%]"
    >
      <div
        class="opacity-70 hover:cursor-pointer"
      >
        backend.dashboard.dataSummaryPanel.title4
      </div>
      <div
        class="mt-2 mb-4 flex"
      >
        <div
          class="mr-4 text-xl font-bold"
        >
          Â¥35,078
        </div>
        <span
          class="border-yellow-500 bg-[#fff7e0] text-yellow-500 dark:bg-[#342c1a] flex items-center rounded-md border  px-2 text-xs"
        >
          <span
            aria-label="fall"
            class="anticon anticon-fall mr-1"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="fall"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M925.9 804l-24-199.2c-.8-6.6-8.9-9.4-13.6-4.7L829 659.5 557.7 388.3c-6.3-6.2-16.4-6.2-22.6 0L433.3 490 156.6 213.3a8.03 8.03 0 00-11.3 0l-45 45.2a8.03 8.03 0 000 11.3L422 591.7c6.2 6.3 16.4 6.3 22.6 0L546.4 490l226.1 226-59.3 59.3a8.01 8.01 0 004.7 13.6l199.2 24c5.1.7 9.5-3.7 8.8-8.9z"
              />
            </svg>
          </span>
          62.7%
        </span>
      </div>
      <div
        class="overflow-hidden text-ellipsis whitespace-nowrap text-xs opacity-70"
      >
        dashboard.dataSummary.panelFailDesc
      </div>
    </div>
  </div>
</div>
`;

exports[`test TotalOrder should match snapshot 1`] = `
<div>
  <div
    class="bg-white dark:bg-darkMode dark:text-white mt-4 max-w-[100%] basis-[100%] rounded-md border border-stone-200 p-5 dark:border-stone-700 md:max-w-[48%] md:basis-[48%] lg:max-w-[32%] lg:basis-[32%] xl:max-w-[23%] xl:basis-[23%]"
  >
    <div
      class="opacity-70 hover:cursor-pointer"
    >
      backend.dashboard.dataSummaryPanel.title3
    </div>
    <div
      class="mt-2 mb-4 flex"
    >
      <div
        class="mr-4 text-xl font-bold"
      >
        17,364
      </div>
      <span
        class="border-yellow-500 bg-[#fff7e0] text-yellow-500 dark:bg-[#342c1a] flex items-center rounded-md border  px-2 text-xs"
      >
        <span
          aria-label="fall"
          class="anticon anticon-fall mr-1"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="fall"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M925.9 804l-24-199.2c-.8-6.6-8.9-9.4-13.6-4.7L829 659.5 557.7 388.3c-6.3-6.2-16.4-6.2-22.6 0L433.3 490 156.6 213.3a8.03 8.03 0 00-11.3 0l-45 45.2a8.03 8.03 0 000 11.3L422 591.7c6.2 6.3 16.4 6.3 22.6 0L546.4 490l226.1 226-59.3 59.3a8.01 8.01 0 004.7 13.6l199.2 24c5.1.7 9.5-3.7 8.8-8.9z"
            />
          </svg>
        </span>
        22.4%
      </span>
    </div>
    <div
      class="overflow-hidden text-ellipsis whitespace-nowrap text-xs opacity-70"
    >
      dashboard.dataSummary.panelFailDesc
    </div>
  </div>
</div>
`;

exports[`test TotalPageViews should match snapshot 1`] = `
<div>
  <div
    class="bg-white dark:bg-darkMode dark:text-white mt-4 max-w-[100%] basis-[100%] rounded-md border border-stone-200 p-5 dark:border-stone-700 md:max-w-[48%] md:basis-[48%] lg:max-w-[32%] lg:basis-[32%] xl:max-w-[23%] xl:basis-[23%]"
  >
    <div
      class="opacity-70 hover:cursor-pointer"
    >
      backend.dashboard.dataSummaryPanel.title1
    </div>
    <div
      class="mt-2 mb-4 flex"
    >
      <div
        class="mr-4 text-xl font-bold"
      >
        43,298
      </div>
      <span
        class="border-primary bg-secondary text-primary flex items-center rounded-md border  px-2 text-xs"
      >
        <span
          aria-label="rise"
          class="anticon anticon-rise mr-1"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="rise"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M917 211.1l-199.2 24c-6.6.8-9.4 8.9-4.7 13.6l59.3 59.3-226 226-101.8-101.7c-6.3-6.3-16.4-6.2-22.6 0L100.3 754.1a8.03 8.03 0 000 11.3l45 45.2c3.1 3.1 8.2 3.1 11.3 0L433.3 534 535 635.7c6.3 6.2 16.4 6.2 22.6 0L829 364.5l59.3 59.3a8.01 8.01 0 0013.6-4.7l24-199.2c.7-5.1-3.7-9.5-8.9-8.8z"
            />
          </svg>
        </span>
        62.7%
      </span>
    </div>
    <div
      class="overflow-hidden text-ellipsis whitespace-nowrap text-xs opacity-70"
    >
      dashboard.dataSummary.panelRiseDesc
    </div>
  </div>
</div>
`;

exports[`test TotalSales should match snapshot 1`] = `
<div>
  <div
    class="bg-white dark:bg-darkMode dark:text-white mt-4 max-w-[100%] basis-[100%] rounded-md border border-stone-200 p-5 dark:border-stone-700 md:max-w-[48%] md:basis-[48%] lg:max-w-[32%] lg:basis-[32%] xl:max-w-[23%] xl:basis-[23%]"
  >
    <div
      class="opacity-70 hover:cursor-pointer"
    >
      backend.dashboard.dataSummaryPanel.title4
    </div>
    <div
      class="mt-2 mb-4 flex"
    >
      <div
        class="mr-4 text-xl font-bold"
      >
        Â¥35,078
      </div>
      <span
        class="border-yellow-500 bg-[#fff7e0] text-yellow-500 dark:bg-[#342c1a] flex items-center rounded-md border  px-2 text-xs"
      >
        <span
          aria-label="fall"
          class="anticon anticon-fall mr-1"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="fall"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M925.9 804l-24-199.2c-.8-6.6-8.9-9.4-13.6-4.7L829 659.5 557.7 388.3c-6.3-6.2-16.4-6.2-22.6 0L433.3 490 156.6 213.3a8.03 8.03 0 00-11.3 0l-45 45.2a8.03 8.03 0 000 11.3L422 591.7c6.2 6.3 16.4 6.3 22.6 0L546.4 490l226.1 226-59.3 59.3a8.01 8.01 0 004.7 13.6l199.2 24c5.1.7 9.5-3.7 8.8-8.9z"
            />
          </svg>
        </span>
        62.7%
      </span>
    </div>
    <div
      class="overflow-hidden text-ellipsis whitespace-nowrap text-xs opacity-70"
    >
      dashboard.dataSummary.panelFailDesc
    </div>
  </div>
</div>
`;

exports[`test TotalUser should match snapshot 1`] = `
<div>
  <div
    class="bg-white dark:bg-darkMode dark:text-white mt-4 max-w-[100%] basis-[100%] rounded-md border border-stone-200 p-5 dark:border-stone-700 md:max-w-[48%] md:basis-[48%] lg:max-w-[32%] lg:basis-[32%] xl:max-w-[23%] xl:basis-[23%]"
  >
    <div
      class="opacity-70 hover:cursor-pointer"
    >
      backend.dashboard.dataSummaryPanel.title2
    </div>
    <div
      class="mt-2 mb-4 flex"
    >
      <div
        class="mr-4 text-xl font-bold"
      >
        76,221
      </div>
      <span
        class="border-primary bg-secondary text-primary flex items-center rounded-md border  px-2 text-xs"
      >
        <span
          aria-label="rise"
          class="anticon anticon-rise mr-1"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="rise"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M917 211.1l-199.2 24c-6.6.8-9.4 8.9-4.7 13.6l59.3 59.3-226 226-101.8-101.7c-6.3-6.3-16.4-6.2-22.6 0L100.3 754.1a8.03 8.03 0 000 11.3l45 45.2c3.1 3.1 8.2 3.1 11.3 0L433.3 534 535 635.7c6.3 6.2 16.4 6.2 22.6 0L829 364.5l59.3 59.3a8.01 8.01 0 0013.6-4.7l24-199.2c.7-5.1-3.7-9.5-8.9-8.8z"
            />
          </svg>
        </span>
        71.7%
      </span>
    </div>
    <div
      class="overflow-hidden text-ellipsis whitespace-nowrap text-xs opacity-70"
    >
      dashboard.dataSummary.panelRiseDesc
    </div>
  </div>
</div>
`;
